### PR TITLE
minimuse use of six module

### LINKED ISSUE #163: minimise use of `six` module
> not needed anymore:
>
> - `from six import string_types` → use standard `str` instead
> - `from six.moves import range` → use standard `range`
> - `from six.moves import map` → use standard `map`
>
> Often unnecessary
>
> - `from six import iteritems, iterkeys` → use the `.items()` function for dicts and series
>
> But perhaps we keep this for another PR, since it requires modifying the code a bit

*(originally posted by @FabianHofmann in #162)*

### PR DESCRIPTION
**Pull request by fneum**
Closes #163 .

## Changes proposed in this Pull Request

`six` is the Python 3 and Python 2 compatibility library https://six.readthedocs.io/. Since we dropped Python 2 support, we should phase out the use of this module.

## Checklist

- ~[ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.~
- ~[ ] Unit tests for new features were added (if applicable).~
- ~[ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).~
- ~[ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.~

### CODE REVIEW

**Comment by fneum** [comment]
I would like to take the current status as a first step, then finalise in another PR.

File: pypsa/pf.py [hunk 1]
```diff
@@ -53,8 +51,7 @@ def imag(X): return np.imag(X.to_numpy())
 def _as_snapshots(network, snapshots):
     if snapshots is None:
         snapshots = network.snapshots
-    if (isinstance(snapshots, six.string_types) or
-        not isinstance(snapshots, (Sequence, pd.Index))):
+    if isinstance(snapshots, str) or not isinstance(snapshots, pd.Index):
         return pd.Index([snapshots])
```
Comment (coroa) [inline pypsa/pf.py:55]: You should not drop the test for being list-like completely here! what if you get a list of snapshots?

https://pandas.pydata.org/pandas-docs/version/1.1/reference/api/pandas.api.types.is_list_like.html should be a good way to go

File: pypsa/components.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 """
 
 
-from six import iteritems, itervalues, iterkeys
+from six import itervalues, iterkeys
```
Comment (coroa) [inline pypsa/components.py:22]: you shouldn't need itervalues or iterkeys anymore, either.

File: pypsa/io.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 """Functions for importing and exporting data.
 """
 
-from six import iteritems, iterkeys, string_types
+from six import iteritems, iterkeys
```
Comment (coroa) [inline pypsa/io.py:19]: Same

File: pypsa/networkclustering.py [hunk 4]
```diff
@@ -23,7 +23,7 @@
 import pandas as pd
 import networkx as nx
 from collections import OrderedDict, namedtuple
-from six.moves import map, range, reduce
+from six.moves import reduce
```
Comment (coroa) [inline pypsa/networkclustering.py:26]: `reduce` lives in `functools` in python3

File: pypsa/opf.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 """
 
 
-from six import iteritems, itervalues, string_types
+from six import iteritems, itervalues
```
Comment (coroa) [inline pypsa/opf.py:23]: Idem

File: test/test_pf_against_pandapower.py [hunk 6]
```diff
@@ -8,13 +8,13 @@
...
+import sys
 
 import pytest
 
 
 @pytest.mark.skipif(
-    six.PY2 and not pandapower_compatible,
+    sys.version_info.major == 2 and not pandapower_compatible,
     reason="Pandapower 2.0.0 dropped support for Python 2.",
 )
```
Comment (coroa) [inline test/test_pf_against_pandapower.py:19]: You can get rid of the whole skipif, that was for not testing pandapower when running python2.

File: pypsa/networkclustering.py [hunk 7]
```diff
@@ -23,7 +23,7 @@
...
 from six import itervalues, iteritems
```
Comment (coroa) [inline pypsa/networkclustering.py:27]: Samesame

**Review by coroa — CHANGES_REQUESTED** [review]

**Pull request merged**